axlearn/cloud/gcp/pathways_utils.py

```diff
@@ -47,20 +49,14 @@
 _PATHWAYS_RESOURCE_MANAGER_CONTAINER_NAME = "pathways-rm"
 # The container name of pathways proxy.
 _PATHWAYS_PROXY_CONTAINER_NAME = "pathways-proxy"
-# The node pool name for pathways-head pod.
-_PATHWAYS_HEAD_NODE_POOL_NAME = "pathways-head"
 # The k8s replicatedJob name for pathways-head pods.
 _PATHWAYS_HEAD_REPLICATED_JOB_NAME = "pathways-head"
 # The k8s replicatedJob name for pathways-worker pods.
 _PATHWAYS_WORKER_REPLICATED_JOB_NAME = "pathways-worker"
 
-
-def get_pathways_head_address(job_name: str) -> str:
-    """Returns the address of the pathways head pod."""
-    # There will be only one pathways-head pod, so it is already 0-0.
-    # First 0 means the first replicatedJob of pathways-head,
-    # the second 0 means the first pod in the replicatedJob.
-    return f"{job_name}-{_PATHWAYS_HEAD_REPLICATED_JOB_NAME}-0-0.{job_name}"
+# Add node-selector for cpu workload to avoid sharing nodes with system services.
+_PATHWAYS_HEAD_NODE_POOL_SELECTOR_KEY = "axlearn/nodepool_type"
```

Comment:
I think we should default to not having any nodeSelector to make this more easily usable for users outside of Apple.